dspace-spring-rest/src/main/java/org/dspace/app/rest/submit/SubmissionService.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.submit;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.atteo.evo.inflector.English;
+import org.dspace.app.rest.converter.BitstreamFormatConverter;
+import org.dspace.app.rest.converter.ResourcePolicyConverter;
+import org.dspace.app.rest.model.BitstreamRest;
+import org.dspace.app.rest.model.CheckSumRest;
+import org.dspace.app.rest.model.MetadataValueRest;
+import org.dspace.app.rest.model.ResourcePolicyRest;
+import org.dspace.app.rest.model.step.UploadBitstreamRest;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.authorize.ResourcePolicy;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Collection;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.WorkspaceItem;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.WorkspaceItemService;
+import org.dspace.core.Context;
+import org.dspace.core.Utils;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.RequestService;
+import org.dspace.services.model.Request;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Service to manipulate in-progress submissions.
+ * 
+ * @author Luigi Andrea Pascarelli (luigiandrea.pascarelli at 4science.it)
+ *
+ */
+@Component
+public class SubmissionService {
```

Comment:
This class seems to contain both "business logic" (e.g. create a workspace item in the default collection if no collection is given) as "rest logic". (In the future) we need to split this into two separate services (one in `dspace-api` and one in `dspace-spring-rest`).

Reply:
Ok, in the future we will decide if to mantain this logic (move this logic in the workspaceitemservice) or manage the creation of workspaceitem in the default mode without superstructure (current implementation of workspaceitemservice#create).